python/semantic_kernel/connectors/ai/open_ai/utils.py

```diff
@@ -103,14 +162,16 @@ async def execute_function_call(kernel: Kernel, function_call: FunctionCall, log
     return str(result)
 
 
+async def execute_tool_call(kernel: Kernel, tool_call: ToolCall) -> str:
+    return await execute_function_call(kernel, tool_call.function)
```

Comment:
should we inverse this so that when we get rid of function_call this does not have to be changed?

Reply:
Makes sense, I think that's a good idea.

Reply:
I've made a change that decouples the execution logic from the specific types of calls, making it easier for us to remove `function_call` in the future.